frontend/main.js

```diff
@@ -0,0 +1,32 @@
+const { app, BrowserWindow } = require('electron');
+const serve = require('electron-serve');
+
+const loadURL = serve({ directory: 'dist' });
+
+let mainWindow;
+
+function createWindow() {
+    mainWindow = new BrowserWindow({
+        width: 800,
+        height: 600,
+        webPreferences: {
+            nodeIntegration: true,
+        },
+    });
+    loadURL(mainWindow);
+}
+
+app.whenReady().then(() => {
+    createWindow();
+    app.on('activate', () => {
+        if (BrowserWindow.getAllWindows().length === 0) {
+            createWindow();
+        }
+    });
```

Comment:
Why is it needed to create window two times?

Reply:
When you apply some hacks (even if they are suggested by the documentation), but are not clear by default, please try to explain it. In this case, please add a comment with the link